lib/prometheus/client/data_stores/direct_file_store.rb

```diff
@@ -68,6 +68,9 @@ def validate_metric_settings(metric_settings)
         end
 
         class MetricStore
+          # Update the filename prefix when the file format changes
+          FILENAME_PREFIX = "metricv2"
```

Comment:
Thought: should the filename_prefix be the `Prometheus::Client::VERSION`? That way, we get incrementing of this value for free, without having to remember to update it.

Another thought: we're already working on the assumption that you start with a clear `/tmp`, so this feels a little out of place by itself, given that a server restart where you keep your old files is already a bad thing and we're not really helping with that by doing this.

I think i'd prefer to not have this second change

Reply:
I had both thoughts too. However, my conclusion was that since the gem isn't cleaning up these files on its own, the burden is currently on the app to do this. I would bet that someone will run into this problem.

See this part of my app: https://github.com/stefansundin/github-activity/commit/936d4d6f1627549390fdf151ab475c448dde12cc#diff-d21913acbe4d1528a62c64cdec6d43d7R11-R14

After this PR, I was going to submit another PR that adds a new flag to DirectFileStore, called `:clean_dir` or something similar, which if true would clean all the directory for you.

A little bit related, that PR would also contain a new method that you can call with a pid, and it would clean up any metrics that used to belong to that process. I would replace this code with a call to that method: https://github.com/stefansundin/github-activity/commit/936d4d6f1627549390fdf151ab475c448dde12cc#diff-3f0c524806c4374c99a509efaffd3e8aR29-R32

Reply:
I opened a second PR, so I can drop this commit from this PR. https://github.com/prometheus/client_ruby/pull/173

Reply:
This commit has been dropped.